doc/md/guides/orms/efcore.md

```diff
@@ -0,0 +1,244 @@
+and its _desired_ state.
+
+In the context of versioned migrations, the current state can be thought of as the database schema that would have
+been created by applying all previous migration scripts.
+
+The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema),
+which is any program that can output a SQL schema definition to stdout.
+To use Atlas with EF Core, users can utilize the [EF Core Atlas Provider](https://github.com/ariga/atlas-provider-ef),
+a .NET tool that can be used to load the schema of an EF Core project into Atlas.
+
+In this guide, we will show how Atlas can be used to automatically plan schema migrations for
+EF Core users.
+
+## Prerequisites
+
+* A local project that uses EF Core for data access.
+
+If you don't have one, you can use [TodoApi](https://github.com/davidfowl/TodoApi) from David Fowler (Microsoft Engineer for 
+things like ASP.NET, Aspire, and NuGet, among others) as a starting point:
+
+```bash
+git clone git@github.com:davidfowl/TodoApi.git
+```
+
+## Using the Atlas EF Core Provider
+
+In this guide, we will use the [Atlas Provider for EF Core](https://github.com/ariga/atlas-provider-ef)
+to automatically plan schema migrations for an EF Core project.
+
+A connection to a live database is optional; it depends on the [Database Provider](https://learn.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli) you are using.
+
+### Installation
+
+<Tabs groupId="operating-systems">
+<TabItem value="win" label="Windows">
+
+Use PowerShell to download the Atlas binary:
+
+```powershell
+Invoke-WebRequest https://release.ariga.io/atlas/atlas-windows-amd64-latest.exe -OutFile atlas.exe
+```
+
+Then add the directory where the Atlas binary is located to the system's PATH environment variable.
+
+</TabItem>
+<TabItem value="mac_linux" label="macOS + Linux">
+
+Install Atlas from macOS or Linux by running:
+
+```bash
+curl -sSf https://atlasgo.sh | sh
+```
+
+</TabItem>
+</Tabs>
+
+See [atlasgo.io](https://atlasgo.io/getting-started#installation) for more installation options.
+
+#### Atlas Provider for EF Core
+
+This package is available on [NuGet](https://www.nuget.org/packages/atlas-ef/).
+
+Navigate to the `TodoApi` folder and create a tool manifest file:
+
+```bash
+dotnet new tool-manifest
+```
+
+Install **atlas-ef** as a local tool using the `dotnet tool` command:
+
+```bash
+dotnet tool install --local atlas-ef
+```
+
+Verify that the tool is installed by running:
+
+```bash
+dotnet atlas-ef --version
+```
+
+### Configuration
+
+Navigate to the `TodoApi` folder and create a new file named `atlas.hcl` with the following contents:
+
+```hcl
+data "external_schema" "ef" {
+  program = [
+    "dotnet",
+    "atlas-ef",
+  ]
+}
+
+env {
+  name = atlas.env
+  src = data.external_schema.ef.url
+  dev = "sqlite://dev?mode=memory" # list of dev dbs can be found here: https://atlasgo.io/concepts/dev-database
```

Comment:
better do a tabs/switch with copyable example for each db

Reply:
Adding more tabs sounds confusing because I’m following the example repo, which uses SQLite. A link to me means "if you want to explore more, play with other databases, you could click on it."